releaseNotes/Microsoft.Featuremanagement.md

```diff
@@ -2,6 +2,50 @@
 
 [Source code][source_code] | [Package (NuGet)][package] | [Samples][samples] | [Product documentation][docs]
 
+## 4.0.0 - September 24, 2024
+
+### Enhancements
+
+#### Variants
+
+```csharp
+Variant variant = await featureManager.GetVariantAsync(MyFeatureFlags.HelpText);
+
+model.Text = variant.Configuration.Value;
+```
+
+Variants are a tool that can be used to surface different variations of a feature to different segments of an audience. Previously, this library only worked with flags. The flags were limited to boolean values, as they are either enabled or disabled. Variants have dynamic values. They can be any valid json value- like a string, int, or complex object.
+
+For more details on Variants, see [here](https://learn.microsoft.com/en-us/azure/azure-app-configuration/feature-management-dotnet-reference#variants).
+
+#### Telemetry
+
+```json
+{
+    "id": "ImageRating",
+    "enabled": true,
+    "telemetry": {
+        "enabled": true
+    }
+}
+```
+
+Telemetry provides observability into flag evaluations, offering insights into which users received specific flag results. This enables more powerful metric analysis, such as experimentation.
+
+For more details on Telemetry, see [here](https://learn.microsoft.com/en-us/azure/azure-app-configuration/feature-management-dotnet-reference#telemetry).
+
+### Migration
+
+**No changes are required** When migrating from 3.5.0 to 4.0.0.
+
+1. In order to use variants or cancellation tokens, change references from `IFeatureManager` to `IVariantFeatureManager`.
+
+### Additional Changes
+
+* Added `IVariantFeatureManager` which is the successor of the existing `IFeatureManager`. It continues to offer the functions of `IFeatureManager`, but offers the new `GetVariantAsync` methods and offers `CancellationToken` on all methods.
```

Comment:
This should be just part of the release note for the "variant feature flag". The `IVariantFeatureManager` is one of many APIs we introduced. The `CancellationToken` support is how the API should be. If you think it's important to call out that `IVariantFeatureManager` is the successor of the existing `IFeatureManager`, call it out in the documentation, which has way more viewers than this release note.

Reply:
It feels separate from the Variants section because of the cancellation / successor part. So I promoted it to it's own section.